syncing/src/test/kotlin/maru/syncing/beaconchain/CLSyncServiceImplTest.kt

```diff
@@ -0,0 +1,493 @@
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getBeaconState(50uL))
+    for (i in 1uL..50uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    // update the sync target to 50 again
+    synced.set(false)
+    reset(beaconChain1, beaconChain2)
+    clSyncService.setSyncTarget(50uL)
+    awaitUntilAsserted { assertThat(synced).isTrue() }
+
+    // Ensure the state remains unchanged
+    assertThat(beaconChain1.getLatestBeaconState().latestBeaconBlockHeader.number).isEqualTo(50uL)
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getBeaconState(50uL))
+    for (i in 1uL..50uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    // Verify that beaconChain for node1 was not updated, and there are no reads on node2 beaconChain
+    verify(beaconChain1, never()).newUpdater()
+    verify(beaconChain2, never()).getSealedBeaconBlocks(any(), any())
+  }
+
+  @Test
+  fun `sync target set to older target returns immediately`() {
+    // sync to block 50
+    val synced = AtomicBoolean(false)
+    clSyncService.setSyncTarget(50uL)
+    clSyncService.onSyncComplete { synced.set(true) }
+    awaitUntilAsserted { assertThat(synced).isTrue() }
+
+    assertThat(beaconChain1.getLatestBeaconState().latestBeaconBlockHeader.number).isEqualTo(50uL)
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getBeaconState(50uL))
+    for (i in 1uL..50uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    // update the sync target to 40, should return immediately
+    synced.set(false)
+    reset(beaconChain1, beaconChain2)
+    clSyncService.setSyncTarget(40uL)
+    awaitUntilAsserted { assertThat(synced).isTrue() }
+
+    // Ensure the state remains unchanged
+    assertThat(beaconChain1.getLatestBeaconState().latestBeaconBlockHeader.number).isEqualTo(50uL)
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getBeaconState(50uL))
+    for (i in 1uL..50uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    // Verify that beaconChain for node1 was not updated, and there are no reads on node2 beaconChain
+    verify(beaconChain1, never()).newUpdater()
+    verify(beaconChain2, never()).getSealedBeaconBlocks(any(), any())
+  }
+
+  @Test
+  fun `chain sync continues to download after stopped is called`() {
```

Comment:
I don't get why this is valid. I think it shouldn't download anything

Reply:
There was a comment that @fluentcrafter left saying it shouldn't matter if we call `start()` before setting the sync target to trigger the sync. I think it responsible considering this a service that we need to call `start()` before it can be used to sync and in the stopped state it won't trigger a sync. Sounds like that's also what you expect. So going to change to back to that behaviour.